nestedtensor/csrc/functions.cpp

```diff
@@ -1,283 +1,170 @@
-    at::Tensor buffer = *input.get_buffer();
-    return NestedTensor(std::move(buffer), std::move(structure));
-  }
-  return NestedTensor(std::move(structure));
+  auto structure = self_data.get_structure();
+  apply([](at::Tensor& t) { at::relu_(t); }, structure);
+  return self;
 }
 
-NestedTensor relu(NestedTensor& input, 
-                  c10::optional<bool> inplace) {
-  if (input.is_contiguous()) {
-    if (inplace.has_value() && inplace.value()) {
-      at::relu_(*input.get_buffer());
-      return input;
-    }
-    return NestedTensor(torch::relu(*input.get_buffer()), input.nested_size());
-  }
-
-  if (inplace.has_value() && inplace.value()) {
-    TensorNode& input_structure = input.get_structure();
-    apply([](at::Tensor& t) { at::relu_(t); }, input_structure);
-    return input;
-  } else {
-    TensorNode& input_structure = input.get_structure();
-    TensorNode res = map([&](at::Tensor t){
-        return torch::relu(t);
-    }, input_structure);
-
-    return NestedTensor(std::move(res));
-  }
+Tensor NestedTensor_dropout(const Tensor& input, double p, bool train) {
+  auto self_impl = get_nested_tensor_impl(input);
+  auto self_data = self_impl->_data;
+  auto structure = self_data.get_structure();
+  auto res =
+      map([&](const at::Tensor t) { return at::dropout(t, p, train); }, structure);
+  return at::detail::make_tensor<NestedTensorImpl>(
+      torch::nested_tensor::NestedTensor(std::move(res)));
 }
 
-void relu_out(NestedTensor& input) {
-  relu(input, true);
+Tensor& NestedTensor_dropout_(Tensor& input, double p, bool train) {
+  auto self_impl = get_nested_tensor_impl(input);
+  auto self_data = self_impl->_data;
+  auto structure = self_data.get_structure();
+  apply(
+      [&](at::Tensor& t) { return at::dropout_(t, p, train); }, structure);
+  return input;
 }
 
-NestedTensor dropout(NestedTensor& input, 
-                     c10::optional<double> p, 
-                     c10::optional<bool> training, 
-                     c10::optional<bool> inplace) {
-  TensorNode input_structure = input.get_structure();
-  TensorNode res = map([&](at::Tensor t){
-      return torch::dropout(t, p.value(), training.value());
-  }, input_structure);
-
-  return NestedTensor(std::move(res));
+Tensor NestedTensor_conv2d(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    int64_t groups) {
+  auto input_impl = get_nested_tensor_impl(input);
+  auto input_data = input_impl->_data;
+  auto structure = input_data.get_structure();
+
+  auto res = map(
+      [&weight, &bias, &stride, &padding, &dilation, groups](at::Tensor t) {
+        return at::convolution(
+                   t.unsqueeze(0),
+                   weight,
+                   bias,
+                   stride,
+                   padding,
+                   dilation,
+                   false,
+                   {{0, 0}},
```

Comment:
If you dont want to use `Conv2dFuncOptions` i would, at least, leave comments what are these values for